JamGame/Assets/Scripts/TileUnion/TileUnionImpl.cs

```diff
@@ -86,7 +94,8 @@ public void SetGridProperties(GridProperties gridProperties)
             this.gridProperties = gridProperties;
         }
 
-        public List<TileImpl> Tiles = new();
+        [SerializeField]
+        private Dictionary<TileImpl, List<TileImpl>> tiles = new();
```

Comment:
Please add a new dictionary like `projectedTiles` to not mix logics for tiles and their projections. Also add a check(editor-only) that tiles with projections cannot be moved by user